### ISSUE TITLE
Improve color contrast within disabled buttons

### ISSUE THREAD

**Issue by fenglugithub** [hackforla/tdm-calculator#2370]
### Overview
Please update the color contrast of disabled buttons in the design system so that users can read button labels more easily.

### Details
Problem: 
- Insufficient contrast between text and background on disabled buttons makes them hard to read.
- Non-compliance with accessibility standards, especially WCAG 2.1 AA minimum contrast requirements.
<details><summary>Screenshots for disabled buttons</summary>

<img width="562" alt="Image" src="https://github.com/user-attachments/assets/2310c24a-5c99-4977-ab82-91c9a6c4cd24" />

<img width="683" alt="Image" src="https://github.com/user-attachments/assets/c6792cb4-38c0-4ecb-b9a6-39221bb6f293" />

</details>

Solution:
- Update color tokens or values to improve the contrast between text/icon and background while maintaining a visually distinct "disabled" state.
- Validate changes using contrast checking tools to ensure they meet or exceed accessibility standards.
- Document updates clearly in the design system with before-and-after examples for development handoff.

### Action Items
- [x] Assign yourself to this issue.
- [x] Move the issue from Prioritized Backlog to In Progress.
- [x] In Figma, move a blank page to the Design section.
- [x] Change the Figma page's title to have the format "#Issue Number | Title | Assignee Name".
- [x] Copy contents from the Figma Template page to this issue's Figma page.
- [x] Include a link to this issue's Figma page in the Figma section below.
- [x] Uses the Figma page to design solutions for this issue.
- [x] Review proposed designs with the design lead. Implement feedback as necessary.
- [x] Use [presentation structure](https://github.com/hackforla/tdm-calculator/wiki/Design:-Presentations) to review proposed designs with the design director. Implement feedback as necessary.
- [ ] Use [presentation structure](https://github.com/hackforla/tdm-calculator/wiki/Design:-Presentations) to review proposed designs with the product, development, and research teams. Implement feedback as necessary.
- [ ] Receive design approval from stakeholders. Implement feedback as necessary.
- [x] Create a Ready for Dev section around the final designs on the Figma page.
- [x] Update the [Figma Design System](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=4041-12655&t=54f7L4Eubz7ppygs-1) with changes resulting from the issue's final design, if applicable.
- [x] Update the [Style Guide Presentation](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/), if applicable.
- [x] Create a [development handoff issue](https://github.com/hackforla/tdm-calculator/issues/new?template=development-handoff-issue.md).
- [x] Move the Figma page to the Development section of the Figma file.
- [x] Add the development issue's number in parentheses at the beginning of the title of the Figma page.
- [x] Include the development handoff issue number in the GitHub section below.
- [x] Reassign any previous assignees to this issue.
- [x] Close this issue.

### Resources
- [WCAG 2.1 AA Guidelines](https://www.w3.org/WAI/WCAG21/quickref/#contrast-minimum)
- [W3C Accessibility Contrast Checker](https://webaim.org/resources/contrastchecker/)
- [Material Design Color Guidelines](https://m3.material.io/styles/color/overview)
- [Style Guide Deck ](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/edit#slide=id.g2a6d1bfeb46_0_103)
- [TDM Site](https://tdm-dev.azurewebsites.net)
- [Staging Deck](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/edit#slide=id.g34bf049f908_1_15)
- Dev Issue - https://github.com/hackforla/tdm-calculator/issues/2524

#### Figma
[Figma section where to find disabled buttons](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=16061-4518&t=SAQKySCp9SBvFQkF-1)

#### Slides
- [Staging Deck](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg)
   - [Staging Deck, slide 2370](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/edit?slide=id.g34bf049f908_1_15#slide=id.g34bf049f908_1_15)
- Release Deck
   - [YYYY-MM-DD Release Deck, slide 2370]

**Comment by ExperimentsInHonesty** [comment]
@fenglugithub I updated the labels and removed the ready for prioritization label. In the future, if you are not sure what labels it gets, you can add the `ready for design lead - any` label and one of them can help you add the labels.

**Comment by JaimeCucalon** [comment]
**Progerss:**
This is ready for Tristan to review.

I've completed my design recommendation. Made edits to the button disabled state to pass WCAG 2.1 AA minimum contrast requirements. I've gone over the design recommendations with @michaelsgruen and he approved the solutions. 

[Here is the Figma page link.
](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=18525-40)

And a screenshot of the design recommendation. 
![Image](https://github.com/user-attachments/assets/7da1db47-427b-4e42-9d8c-b422ba84881f)

Blockers: Waiting on review before applying these styling recommendations to Arrow buttons.
Availability: 6 hours
ETA: May 9

**Comment by thetanmancan** [comment]
Hi @JaimeCucalon. I really like the work you did here. It works well and is very nicely organized. I have a few notes.

- There is a problem that #2297 is designed to fix but hasn't yet: "Gray" in the Figma library is different than what is written in the design system. In the library it's `#5E5E5E`, and in the design system it's `#939393`. You're using the library version, which is darker. This will be something to address in #2297.
- I like the work you did with the revised disabled warning button. I agree that V1 is preferable. The good news is that, even if it's not as accessible as we'd like, I don't think we have a disabled warning button anywhere in our system, so it shouldn't matter, unless we end up needing to design one for a future function or modal.
- I agree that adding the tertiary color to our design system makes sense. Please add that as part of this issue.
- Do you think it's worth adding the new disabled colors to the design system as well? We don't have to if they are all modified versions of the colors we already have, but I think only if they're all modified in the same way. If they each have a different rule for their modification, I would think adding them to the design system would be useful.
- I agree that removing the drop shadow for disabled buttons makes sense. That's going to require a new issue, though. I'll make note of that.

**Comment by thetanmancan** [comment]
Hi @JaimeCucalon. After adding the tertiary and disabled colors to our design system and removing the drop shadow from the disabled buttons, please remember to write a development handoff issue that applies those changes to our system. I'll still be here on Friday if you have any questions about how to do that, or if you end up writing the issue after I leave, feel free to reach out to someone from the original TDM team, since I went over how to write a handoff issue with them on Wednesday. Thanks!

**Comment by lola3736** [comment]
@JaimeCucalon 
Please provide update

<details><summary>Instructions</summary>

1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (if necessary): "Add any pictures that will help illustrate what you are working on."
</details> 

1. Progress: 
2. Blockers: 
3. Availability:
4. ETA:
5. Pictures (if necessary):

**Comment by JaimeCucalon** [comment]
**Progress:**
- Added tertiary and disabled colors to Figma page for review. Added description and made a small mention of inconsistencies for gray colors. 
- Updated pagination button with new disabled styling/colors

**Blockers:**
- Waiting on visual review with Bonnie to make final color additions to Full Design System. And before updating components in system. 

![Image](https://github.com/user-attachments/assets/12210355-8986-4356-9f38-4b5125f81797)

![Image](https://github.com/user-attachments/assets/c67e4f5d-944d-4858-8ac7-60c707b5cad7)

**Comment by ExperimentsInHonesty** [comment]
[Issue tab that references the color 5E5E5E](https://github.com/hackforla/tdm-calculator/issues?q=is%3Aissue%20state%3Aopen%205E5E5E%20)

[Issue tab that references the color 939393](https://github.com/hackforla/tdm-calculator/issues?q=is%3Aissue%20state%3Aopen%20939393)

**Comment by JaimeCucalon** [comment]
**Design for this issue was completed and approved by Bonnie on May 7, 2025.**
- The design system has been updated with new button and pagination component. [Link to updated components](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=16061-4518&t=k4oEhlElDHOFf3wo-4)
- New colors and annotations have added to the design system as well. [Link to updated color palette ](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=16061-5221&t=k4oEhlElDHOFf3wo-4)
- Updated the [Style Guide Presentation](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/edit#slide=id.g2a6d1bfeb46_0_103)
- Set revised components as "**ready for dev**" in Figma page. Including some notes for developers. [Link to components ready for dev ](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=22284-111883&t=k4oEhlElDHOFf3wo-4)

- 
- Leaving a separate note related to inconsistencies in the design system "gray" color for #2297 .The revisions made to the button and pagination component disabled text use the gray color #5E5E5E. This color has been set as "Gray" in the Figma file, yet the color palette in the full design system lists "Gray" as #939393.

**Comment by JaimeCucalon** [comment]
Dev Issue created. [#2524](https://github.com/hackforla/tdm-calculator/issues/2524)

**Comment by ExperimentsInHonesty** [comment]
@JaimeCucalon Thank you for your work on this issue 😎
